02_find_user_country_by_ip/src/loadData.ts

```diff
@@ -0,0 +1,25 @@
+import fs from 'fs';
+
+export type IPRange = {
+    start: number;
+    end: number;
+    countryCode: string;
+    countryName: string;
+}
+
+export const loadIPRanges = () => {
+    const lines = fs.readFileSync('IP2LOCATION-LITE-DB1.CSV', 'utf8').split('\n');
+    const ipRanges: IPRange[] = new Array(lines.length);
+    for (let i = 0; i < lines.length; i++) {
+        if (lines[i].length > 0) {
+            let parts = lines[i].split(',');
+            ipRanges[i] = {
+                start: Number(parts[0].trim().slice(1, -1)),
+                end: Number(parts[1].trim().slice(1, -1)),
+                countryCode: parts[2].trim().slice(1, -1),
+                countryName: parts[3].trim().slice(1, -1)
+            }
+        }
+    }
+    return ipRanges;
+}
```

Comment:
We can rewrite this bunch of code with reduce:


const formatNumberRange = (ipPart: string, leftBorder: number = 1, rightBorder: number = -1 ) => Number(ipPart.trim().slice(leftBorder, rightBorder)),


export const loadIPRanges = () => {
    const lines = fs.readFileSync('IP2LOCATION-LITE-DB1.CSV', 'utf8').split('\n');
    
    return lines.reduce((ipRanges, line) => {
        if (line.length > 0) {
            const parts = line.split(',');
            ipRanges.push({
                start: formatNumberRange(parts[0]),
                end: formatNumberRange(parts[1]),
                countryCode: formatNumberRange(parts[2]),
                countryName: formatNumberRange(parts[3]),
            });
        }
        return ipRanges;
    }, []);
}

Reply:
Thank you, it really is!